subsys/nrf_compress/lzma/Lzma2Dec.c

```diff
@@ -240,6 +250,9 @@ SRes Lzma2Dec_DecodeToDic(CLzma2Dec *p, SizeT dicLimit, const Byte *src, SizeT *
 				}
 
 				LzmaDec_UpdateWithUncompressed(&p->decoder, src, inCur);
+				if (p->state == LZMA2_STATE_ERROR) {
+					continue;
+				}
```

Comment:
this change doesn't even make sense, neither function above changes this, and I would rather the original lzma code flow is not altered

Reply:
This addition here is not going to be used, the only place in the code above where p->state is updated is line 216, which has this:
```
			p->state = Lzma2Dec_UpdateState(p, *src++);
			if (dicPos == dicLimit && p->state != LZMA2_STATE_FINISHED) {
				break;
			}
			continue;
```

So it either breaks or continues, if it breaks then this whole loop is finished, if it continues then it goes back to the state which then has:
```
	while (p->state != LZMA2_STATE_ERROR) {
```
So what is the point in this code addition when it does nothing?